cypress/component/advanced/react-router-v6/in-memory-spec.js

```diff
@@ -0,0 +1,32 @@
+/// <reference types="cypress" />
+import React from 'react'
+import { mount } from 'cypress-react-unit-test'
+import { App } from './app.jsx'
+import { MemoryRouter } from 'react-router-dom'
+
+it('navigates through the link without changing url', () => {
+  cy.viewport(600, 300)
+  mount(
+    <MemoryRouter
+      initialEntries={['/about', '/two', { pathname: '/three' }]}
+      initialIndex={0}
+    >
+      <App />
+    </MemoryRouter>,
+  )
+
+  // we are mocking the initial open route with `initialIndex`
+  cy.contains('h2', 'About')
+
+  // Go to home route
+  cy.contains('a', 'Home').click()
+
+  cy.contains('h2', 'Home')
+  cy.location('pathname').should('not.equal', '/')
```

Comment:
so is it changing the location or not? And does MemoryRouter have some state we could check in our test directly to confirm navigation?

Reply:
No it's not. And it's the main reason of using it, it is not changing url directly and allows to mock browser history. 

We are testing that it works by asserting `h2` to contain proper text. Home or About.

Reply:
I mean the `cy.location` in this case is extremely confusing since it returns the spec filename :)

<img width="619" alt="Screen Shot 2020-09-16 at 1 04 57 PM" src="https://user-images.githubusercontent.com/2212006/93369352-42ff7780-f81d-11ea-8013-4c67cecd23c5.png">

Reply:
Yes because the original href is spec path. This confirms that url in browser not changing but tests works.

Reply:
Right - but the assertion can pass for the wrong reason (it could not go to the about page, and yet no match `/`). See how I changed it to have positive assertions and added readme and screenshot gif explaining what is going on.

Reply:
I would not say it is the best way to change it. Only for example, from the user perspective – in tests with memory router we care about what rendered, right? We are checking it with assertions that shows in `h2`. 

Assertion for file name could also be a variant, but easy to break :)

Reply:
good point, I moved the location assertion after the component text assertion. But other than that - yes, the user should in this case only assert the right component is rendered in the DOM, not the URL. I did URL mostly to confirm that the URL is NOT changing when using memory router

Reply:
Maybe also a good example would be to show how `cy.visit` works. I hope it would not have any effect